attributes/security.rb

```diff
@@ -17,10 +17,20 @@
 # limitations under the License.
 #
 
-# COOK-74
+# COOK-74 and COOK-116
 if node['cdap'].key?('cdap_site') && node['cdap']['cdap_site'].key?('security.server.ssl.enabled') &&
+   !node['cdap']['cdap_site'].key?('ssl.external.enabled') &&
    !node['cdap']['cdap_site'].key?('ssl.enabled')
   default['cdap']['cdap_site']['ssl.enabled'] = node['cdap']['cdap_site']['security.server.ssl.enabled']
+  default['cdap']['cdap_site']['ssl.external.enabled'] = node['cdap']['cdap_site']['security.server.ssl.enabled']
+elsif node['cdap'].key?('cdap_site') && node['cdap']['cdap_site'].key?('ssl.enabled') &&
+      !node['cdap']['cdap_site'].key?('security.server.ssl.enabled') &&
+      !node['cdap']['cdap_site'].key?('ssl.external.enabled')
+  default['cdap']['cdap_site']['security.server.ssl.enabled'] = node['cdap']['cdap_site']['ssl.enabled']
+  default['cdap']['cdap_site']['ssl.external.enabled'] = node['cdap']['cdap_site']['ssl.enabled']
+else
```

Comment:
this should be ``elsif node['cdap'].key?('cdap_site') && node['cdap']['cdap_site'].key?('ssl.external.enabled')``.  otherwise in the next lines it'll add ``nil`` as a new value which could cause problems later